### PR TITLE
[10.0][FIX] remove carrier_id when the shipping address is modified

### PR DESCRIPTION
**Pull request by bguillot**
If the shipping address is modified in the cart, we remove the carrier.

### CODE REVIEW

File: shopinvader_delivery_carrier/services/cart.py [hunk 1]
```diff
@@ -133,6 +133,13 @@ def _delete_item(self, cart, params):
         self._unset_carrier(cart)
         return res
 
+    def update(self, **params):
+        if params.get('shipping', {}).get('address'):
```
Comment (lmignon) [inline shopinvader_delivery_carrier/services/cart.py]: @bguillot are we sure that we always receive all the information of the SO? Is that the expected behavior?

**Comment by rousseldenis** [comment]
@bguillot What's the status of this ?

**Comment by rousseldenis** [comment]
@bguillot

File: shopinvader_delivery_carrier/services/cart.py (hunk 1 above)
Comment (sebastienbeau) [reply shopinvader_delivery_carrier/services/cart.py]: yes it's the format for updating the shipping address
I have added a test.

**Comment by sebastienbeau** [comment]
@rousseldenis I have updated the PR. The fix is still needed
If we update the delivery address (going back in the cart process), the carrier was not unset.

**Comment by sebastienbeau** [comment]
@rousseldenis @Cedric-Pigeon  it's ok for you?

**Review by Cedric-Pigeon — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**